### PR TITLE
test: validate secondary endpoint client failure

### PR DESCRIPTION
**Pull request by jaer-tsun**
**Reason for Change**:
Add a test to validate that endpoint will not be created if SecondaryEndpointClient fails

Passing NIC Type var from CNS response down instead of hardcoding

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: network/endpoint_test.go [hunk 1]
```diff
@@ -262,6 +262,37 @@ var _ = Describe("Test Endpoint", func() {
...
+				}
+				secondaryEpInfo := &EndpointInfo{
+					MacAddress: netio.BadHwAddr, // mock netlink will fail to set link state on bad eth
+					NICType:    cns.DelegatedVMNIC,
+					Routes:     []RouteInfo{{Dst: *ipnet}},
+				}
+				ep, err := nw.newEndpointImpl(nil, netlink.NewMockNetlink(false, ""), platform.NewMockExecClient(false),
+					netio.NewMockNetIO(false, 0), nil, NewMockNamespaceClient(), []*EndpointInfo{epInfo, secondaryEpInfo})
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("SecondaryEndpointClient Error: Mock Netlink Error"))
```
Comment (vipul-21) [inline network/endpoint_test.go:286]: nit: we can reuse the variable for error `netlink.ErrorMockNetlink`.

**Review by vipul-21 — APPROVED** [review]

File: network/endpoint_test.go (hunk 1 above)
Comment (jaer-tsun) [reply network/endpoint_test.go:286]: sure, updated

**Review by vipul-21 — DISMISSED** [review]

File: network/endpoint_test.go [hunk 2]
```diff
@@ -262,6 +262,37 @@ var _ = Describe("Test Endpoint", func() {
...
+					netio.NewMockNetIO(false, 0), nil, NewMockNamespaceClient(), []*EndpointInfo{epInfo, secondaryEpInfo})
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("SecondaryEndpointClient Error: " + netlink.ErrorMockNetlink.Error()))
+				Expect(ep).To(BeNil())
+
+				secondaryEpInfo.MacAddress = netio.HwAddr
+				ep, err = nw.newEndpointImpl(nil, netlink.NewMockNetlink(false, ""), platform.NewMockExecClient(false),
+					netio.NewMockNetIO(false, 0), nil, NewMockNamespaceClient(), []*EndpointInfo{epInfo, secondaryEpInfo})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ep.Id).To(Equal(epInfo.Id))
```
Comment (nddq) [inline network/endpoint_test.go:293]: If this test is testing for secondary endpoint add failed, then I don't think we should be testing for when it is success here. Maybe have a separate test for this?

**Review by nddq — CHANGES_REQUESTED** [review]
Just a nit comment on the UT, just wondering, will the secondary endpoint client failed only on bad MAC address, or would there be other scenarios that could cause this failure also?

File: network/endpoint_test.go (hunk 2 above)
Comment (jaer-tsun) [reply network/endpoint_test.go:294]: sure, split

**Review by nddq — APPROVED** [review]

**Review by vipul-21 — APPROVED** [review]

**Pull request merged**